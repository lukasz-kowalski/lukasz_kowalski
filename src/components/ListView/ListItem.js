import React from 'react';
import { Link } from 'react-router-dom';

class ListItem extends React.Component {
  handleDelete = () => {
    this.props.deleteCharacter(this.props.character.id);
  }

  render() {
    const { character } = this.props;
    return (
      <tr key={character.name}>
        <th scope="row">{character.id}</th>
        <td>{character.name}</td>
        <td>{character.species}</td>
        <td>{character.gender}</td>
        <td>{character.homeworld}</td>
        <td>
          <div className="btn-group btn-group-sm" role="group" aria-label="Actions">
            <button type="button" className="btn btn-secondary">
              <Link 
                to={{
                  pathname: `/edit-character/${character.id}`,
                  state: character
                }}
              >
                <i className="fa fa-pencil" aria-hidden="true"/> Edit
              </Link>
            </button>
            <button type="button" className="btn btn-danger" onClick={this.handleDelete}>
              <i className="fa fa-trash-o" aria-hidden="true"/> Remove
            </button>
          </div>
        </td>
      </tr>
    );
  }
}

export default ListItem;
