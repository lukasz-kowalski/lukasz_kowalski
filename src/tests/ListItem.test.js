import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../components/ListView/ListItem';

let wrapper;

const character = {
  name: 'Test',
  species: 'Martian',
  gender: 'female',
  homeworld: ''
}

beforeEach(() => {
  wrapper = shallow(
  <ListItem
    character={character}
  />);
})

test('should successful render ListItem', () => {
  expect(wrapper).toMatchSnapshot();
})
