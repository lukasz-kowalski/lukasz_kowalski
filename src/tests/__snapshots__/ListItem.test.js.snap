// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render ListItem 1`] = `
<tr
  key="Test"
>
  <th
    scope="row"
  />
  <td>
    Test
  </td>
  <td>
    Martian
  </td>
  <td>
    female
  </td>
  <td />
  <td>
    <div
      aria-label="Actions"
      className="btn-group btn-group-sm"
      role="group"
    >
      <button
        className="btn btn-secondary"
        type="button"
      >
        <Link
          replace={false}
          to={
            Object {
              "pathname": "/edit-character/undefined",
              "state": Object {
                "gender": "female",
                "homeworld": "",
                "name": "Test",
                "species": "Martian",
              },
            }
          }
        >
          <i
            aria-hidden="true"
            className="fa fa-pencil"
          />
           Edit
        </Link>
      </button>
      <button
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-trash-o"
        />
         Remove
      </button>
    </div>
  </td>
</tr>
`;
